flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/debezium/DebeziumJsonSerializationSchema.java

```diff
@@ -138,81 +185,202 @@ public byte[] serialize(Event event) {
+        SchemaBuilder schemaBuilder = SchemaBuilder.struct();
+        SchemaBuilder beforeBuilder = SchemaBuilder.struct();
+        SchemaBuilder afterBuilder = SchemaBuilder.struct();
+        for (Column column : columns) {
+            String columnName = column.getName();
+            org.apache.flink.cdc.common.types.DataType columnType = column.getType();
+            final SchemaBuilder field;
+            switch (columnType.getTypeRoot()) {
+                case TINYINT:
+                case SMALLINT:
+                    field = SchemaBuilder.int16();
+                    break;
+                case INTEGER:
+                    field = SchemaBuilder.int32();
+                    break;
+                case BIGINT:
+                    field = SchemaBuilder.int64();
+                    break;
+                case DECIMAL:
+                    final int decimalPrecision = ((DecimalType) columnType).getPrecision();
+                    final int decimalScale = ((DecimalType) columnType).getScale();
+                    field =
+                            Decimal.builder(decimalScale)
+                                    .parameter(
+                                            "connect.decimal.precision",
+                                            String.valueOf(decimalPrecision));
+                    break;
+                case BOOLEAN:
+                    field = SchemaBuilder.bool();
+                    break;
+                case FLOAT:
+                case DOUBLE:
+                    field = SchemaBuilder.float64();
+                    break;
+                case DATE:
+                    field = SchemaBuilder.int32().name(Date.SCHEMA_NAME).version(1);
+                    break;
+                case TIME_WITHOUT_TIME_ZONE:
+                    field = SchemaBuilder.int64().name(MicroTime.SCHEMA_NAME).version(1);
+                    break;
+                case TIMESTAMP_WITHOUT_TIME_ZONE:
+                case TIMESTAMP_WITH_TIME_ZONE:
+                    field = SchemaBuilder.string().name(Timestamp.SCHEMA_NAME).version(1);
+                    break;
+                case BINARY:
+                    field =
+                            SchemaBuilder.bytes()
+                                    .name(Bits.LOGICAL_NAME)
+                                    .parameter(
+                                            Bits.LENGTH_FIELD,
+                                            Integer.toString(
+                                                    org.apache.flink.cdc.common.types.DataTypes
+                                                            .getLength(columnType)
+                                                            .orElse(0)))
+                                    .version(1);
+                    break;
+                case CHAR:
+                case VARCHAR:
+                case VARBINARY:
+                default:
+                    field = SchemaBuilder.string();
```

Comment:
This includes ARRAY/MAP/ROW, There may be some issues if all they were converted to `SchemaBuilder.string`.
You can check if there is a better type, and if not, it is also acceptable.

Reply:
debezium-json does not seem to receive ARRAY/MAP/ROW type data because the mysql does not support this type